source/MaterialXContrib/MaterialXNode/MaterialXNode.cpp

```diff
@@ -70,28 +71,26 @@ MStatus MaterialXNode::initialize()
 	MFnStringData stringData;
 
 	MObject theString = stringData.create();
+
 	DOCUMENT_ATTRIBUTE = typedAttr.create(DOCUMENT_ATTRIBUTE_LONG_NAME, DOCUMENT_ATTRIBUTE_SHORT_NAME, MFnData::kString, theString);
-	CHECK_MSTATUS(typedAttr.setStorable(true));
-	CHECK_MSTATUS(typedAttr.setReadable(true));
-	CHECK_MSTATUS(typedAttr.setInternal(true));
-	CHECK_MSTATUS(typedAttr.setCached(false));
-	CHECK_MSTATUS(typedAttr.setHidden(true));
+    CHECK_MSTATUS(typedAttr.setInternal(true));
+    CHECK_MSTATUS(typedAttr.setKeyable(false));
+    CHECK_MSTATUS(typedAttr.setAffectsAppearance(true));
+    CHECK_MSTATUS(typedAttr.setUsedAsFilename(true));
```

Comment:
It's now a filename and it's not hidden. Is the long name "document" specific enough though? Should it perhaps be "documentPath" or even "documentFilePath" instead?

Reply:
It's supposed to be hidden as no one should actually be able to set this or the element long term. As I mentioned before it's erroneous to assume documents only come from file. e.g if the editor sends us a document update in memory. I'd leave it as document so it's just a string attribute which could contain in memory contents or a filename.
Maybe for editing we only allow the command to set the document+element names? TBD.

Reply:
After Eric's clarification, let's go with public attributes.